docs/index.rst

```diff
@@ -1,10 +1,9 @@
-##################################
-Qiskit IBM Q Account documentation
-##################################
+####################################
+Qiskit IBM Q Provider documentation
```

Comment:
I think it is now IBM Quantum Provider -- @nonhermitian do you know?

Reply:
I debated about this. I thought since it's `ibmq-provider`, it should be `IBM Q Provider`. But I'm open to suggestions.

Reply:
I would like to call it IQX provider, but lets leave it as IBM Quantum for now.